### DISCUSSION TITLE
Brew hangs for a LONG time when run

### DISCUSSION THREAD

**Discussion by TangoOversway** [Homebrew/discussions#2976] [Everyday usage]
I tried to use brew to install libdvdcss. It updated stuff, then just sat there for over an hour:

```
[22-02-16 11:20:31 hal@arda ~] $ brew install libdvdcss
Running `brew update --preinstall`...
==> Auto-updated Homebrew!
Updated 2 taps (homebrew/core and homebrew/cask).
==> New Formulae
alpscore                 brev                     fb303                    jless                    kyverno                  numdiff                  rure                     tagref                   terraform-lsp            vkectl                   xidel
asyncapi                 csview                   ffmpeg@4                 kopia                    linode-cli               oh-my-posh               spago                    tarlz                    tfschema                 webp-pixbuf-loader       yamale
bk                       esphome                  inotify-tools            kubescape                linux-headers@5.16       ruby@3.0                 sshs                     terminalimageviewer      usbutils                 websocketpp
==> Updated Formulae
Updated 722 formulae.
==> New Casks
arctype              devtoys              hackolade            hostsx               imagex               inkstitch            manyverse            polypane             quit-all             thinlinc-client      threema-work         trex                 ujam
==> Updated Casks
Updated 611 casks.
==> Deleted Casks
finisher-voodoo                    hoststool                          jiohome                            jsui                               knox                               liya                               powerword                          telavox-flow
```

After a few hours, I went back, saw that brew had stopped and nothing was happening. I used CTRL-C and stopped it and restarted it again:

```
[22-02-17 2:24:35 hal@sirhokus ~] $ brew install libdvdcss
Running `brew update --preinstall`...
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> Updated Formulae
Updated 4 formulae.
```
And it's stopped there (with a blank line after the last response) for another 90 minutes.

What's going on and what do I need to do to fix it?

**Comment by SMillerDev** [comment]
Try doing `brew update-reset` and see if the problem persists. If it does could you provide me with the output of `brew config` and `brew doctor`?

**Reply by TangoOversway** [reply]
Output from the command:
```
[22-02-17 3:59:53 hal@arda ~] $ brew install libdvdcss
Running `brew update --preinstall`...
==> Auto-updated Homebrew!
Updated 2 taps (homebrew/core and homebrew/cask).
==> Updated Formulae
Updated 2 formulae.
==> Updated Casks
Updated 21 casks.

^C
```
It just hung - about to go to sleep, so didn't want to wait for an hour or more to see what it would do.
```
[22-02-17 4:19:15 hal@arda ~] $ brew config
HOMEBREW_VERSION: 3.3.15-56-gfbf94fe
ORIGIN: https://github.com/Homebrew/brew
HEAD: fbf94fe2c0a50be34f938e76ea6ce0dd4d0f9a3f
Last commit: 6 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: d4372e36da3a048cc11307ddcbd74025b14a834e
Core tap last commit: 8 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.zYKtZz8uN6/org.macosforge.xquartz:0
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.8 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: quad-core 64-bit haswell
```
And at that point it hung again.

`brew doctor` just seems to sit there and does nothing.

**Reply by SMillerDev** [reply]
Did you try `brew update-reset`?

**Reply by TangoOversway** [reply]
Yes. Sorry I wasn't clear that I had, but I did and it finished:
```
[22-02-17 3:59:14 hal@arda ~] $ brew update-reset
==> Fetching /usr/local/Homebrew...
remote: Enumerating objects: 1, done.
remote: Total 1 (delta 0), reused 0 (delta 0), pack-reused 1
Unpacking objects: 100% (1/1), done.
From https://github.com/Homebrew/brew
 * [new branch]          fxcoudert-patch-1 -> origin/fxcoudert-patch-1

==> Resetting /usr/local/Homebrew...
Branch 'master' set up to track remote branch 'master' from 'origin'.
Switched to and reset branch 'master'
Your branch is up to date with 'origin/master'.

==> Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask...
remote: Enumerating objects: 40, done.
remote: Counting objects: 100% (38/38), done.
remote: Compressing objects: 100% (27/27), done.
remote: Total 40 (delta 27), reused 11 (delta 11), pack-reused 2
Unpacking objects: 100% (40/40), done.
From https://github.com/Homebrew/homebrew-cask
   50883701da..1f1c9e6e2d  master     -> origin/master

==> Resetting /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask...
Branch 'master' set up to track remote branch 'master' from 'origin'.
Reset branch 'master'
Your branch is up to date with 'origin/master'.

==> Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core...
remote: Enumerating objects: 94, done.
remote: Counting objects: 100% (86/86), done.
remote: Compressing objects: 100% (30/30), done.
remote: Total 94 (delta 64), reused 75 (delta 56), pack-reused 8
Unpacking objects: 100% (94/94), done.
From https://github.com/Homebrew/homebrew-core
   3d7ab54322d..6e47902f33f  master                   -> origin/master
 * [new branch]              recreate-macos-runners   -> origin/recreate-macos-runners
 * [new branch]              remove-disabled-formulae -> origin/remove-disabled-formulae

==> Resetting /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core...
Branch 'master' set up to track remote branch 'master' from 'origin'.
Reset branch 'master'
Your branch is up to date with 'origin/master'.
```

**Reply by SMillerDev** [reply]
Does `ruby -e "print 'hello'"` also hang?

**Reply by TangoOversway** [reply]
No, that did not hang, but it did tell me that a couple directories in my path were world-writeable, which I had not realized. (One, for example, was one of my own Python development directories.) I changed each one to make them more secure, then that command worked fine. After fixing those directories, I tried `brew reset` and got this:

[22-02-17 12:51:24 hal@arda ~] $ brew update-reset
==> Fetching /usr/local/Homebrew...
remote: Enumerating objects: 6, done.
remote: Counting objects: 100% (6/6), done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 6 (delta 4), reused 6 (delta 4), pack-reused 0
Unpacking objects: 100% (6/6), done.
From https://github.com/Homebrew/brew
   fbf94fe2c..bdf25a57c  master     -> origin/master

==> Resetting /usr/local/Homebrew...
Branch 'master' set up to track remote branch 'master' from 'origin'.
Reset branch 'master'
Your branch is up to date with 'origin/master'.

==> Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask...
remote: Enumerating objects: 32, done.
remote: Counting objects: 100% (24/24), done.
remote: Total 32 (delta 24), reused 24 (delta 24), pack-reused 8
Unpacking objects: 100% (32/32), done.
From https://github.com/Homebrew/homebrew-cask
   812ebaeb86..ca26efe2e4  master     -> origin/master

==> Resetting /usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask...
Branch 'master' set up to track remote branch 'master' from 'origin'.
Reset branch 'master'
Your branch is up to date with 'origin/master'.

==> Fetching /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core...
remote: Enumerating objects: 144, done.
remote: Counting objects: 100% (132/132), done.
remote: Compressing objects: 100% (51/51), done.
remote: Total 144 (delta 94), reused 117 (delta 81), pack-reused 12
Receiving objects: 100% (144/144), 264.65 KiB | 2.38 MiB/s, done.
Resolving deltas: 100% (94/94), completed with 15 local objects.
From https://github.com/Homebrew/homebrew-core
   d4372e36da3..c0c7622bd9d  master     -> origin/master

==> Resetting /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core...
Branch 'master' set up to track remote branch 'master' from 'origin'.
Reset branch 'master'
Your branch is up to date with 'origin/master'.

Then I tried ` brew install libdvdcss` again and it froze after I hit <return>. No response on anything.

**Reply by littlecapa** [reply]
Hi TangoOversway!

Have you been able to resolve your problem? I am facing a similar challenge. Maybe your solution could help me as well.

**Reply by TangoOversway** [reply]
> Hi TangoOversway!
> 
> Have you been able to resolve your problem? I am facing a similar challenge. Maybe your solution could help me as well.

No. Never had any success. I had to drop it and move on to other projects and haven't been able to come back to it. So Brew has been useless to me for about a year now. Haven't heard anything new (as you can see) and no response to my last info on what I was getting.

I think I'm going to have to find a way to get a list of Brew packages, uninstall it, re-install, then reinstall all the packages.

This is why I have never been big on Brew, MacPorts, or anything similar. At some point they always break. (Plus some usurp other programs - took me a few days, at one point, with MacPorts or something similar, to find out why Python packages I was adding were not showing up - it was because they completely usurped my Python install.) Things like this have just discouraged me from this kind of program.

**Reply by SMillerDev** [reply]
> I think I'm going to have to find a way to get a list of Brew packages, uninstall it, re-install, then reinstall all the packages.

`brew bundle dump` can get you a list of all packages
